app/src/androidTest/java/org/mozilla/focus/activity/BrowsingIntentTest.java

```diff
@@ -153,7 +140,8 @@ private void sendBrowsingIntent() {
         Intent intent = new Intent();
         intent.setAction(Intent.ACTION_VIEW);
         intent.setData(Uri.parse(TARGET_URL_SITE_1));
-        intent.setPackage(InstrumentationRegistry.getInstrumentation().getTargetContext().getPackageName());
-        activityTestRule.launchActivity(intent);
+        final Context targetContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        intent.setPackage(targetContext.getPackageName());
+        targetContext.startActivity(intent);
```

Comment:
Just curious about why use `targetContext.startActivity(intent)` instead of `activityTestRule.launchActivity(intent)`?

Reply:
cause activityTestRule has already called launchActivity at this time. (see receivedBrowsingIntent_tabIsCreated). Here we only want to simulate third-party app sending browsing url intent to rocket.